### PR TITLE
remove additional and useless loop for each relation file when do ptrack_get_pagemapset query

### PR DESCRIPTION
**Pull request by vegebird**
remove additional and useless loop for each relation file when do ptrack_get_pagemapset query reported by https://github.com/postgrespro/ptrack/issues/12

### CODE REVIEW

**Comment by vegebird** [comment]
@ololobus Hello Alexey, could you please have a look at this PR ?
In rainy day scenarios, ptrack_get_pagemapset() may return error pagecount/pagemap for the changed relation files when do additional ptrack_map->entries[hash] checker and hash collision happens.

File: ptrack.c [hunk 1]
```diff
@@ -368,6 +368,14 @@ ptrack_filelist_getnext(PtScanCtx * ctx)
 		return ptrack_filelist_getnext(ctx);
 	}
 
+	if (fst.st_size == 0)
+	{
+		elog(WARNING, "ptrack: skip empty file %s", fullpath);
+
+		/* But try the next one */
+		return ptrack_filelist_getnext(ctx);
+	}
```
Comment (funny-falcon) [inline ptrack.c:377]: А правда нужно пропускать?

File: ptrack.c [hunk 2]
```diff
@@ -494,7 +502,7 @@ ptrack_get_pagemapset(PG_FUNCTION_ARGS)
 		XLogRecPtr	update_lsn2;
 
 		/* Stop traversal if there are no more segments */
-		if (ctx->bid.blocknum > ctx->relsize)
+		if (ctx->bid.blocknum + 1 > ctx->relsize)
```
Comment (funny-falcon) [inline ptrack.c:505]: why not `ctx->bid.blocknum >= ctx->relsize` ?

**Review by funny-falcon — APPROVED** [review]
Well, I approve.
Please, accept suggestion on condition.

**Pull request merged**